Sources/StreamChatUI/Common Views/ChatChannelCreateNewButton.swift

```diff
@@ -5,16 +5,19 @@
 import StreamChat
 import UIKit
 
-open class ChatChannelCreateNewButton<ExtraData: ExtraDataTypes>: Button, UIConfigProvider {
-    // MARK: - Overrides
-    
+/// A Button subclass that should be used for creating new channels.
+public typealias ChatChannelCreateNewButton = _ChatChannelCreateNewButton<NoExtraData>
+
+/// A Button subclass that should be used for creating new channels.
+open class _ChatChannelCreateNewButton<ExtraData: ExtraDataTypes>: Button, UIConfigProvider {
     override public func defaultAppearance() {
-        defaultIntrinsicContentSize = .init(width: 44, height: 44)
+        super.defaultAppearance()
         setImage(uiConfig.images.newChat, for: .normal)
     }
-
-    open var defaultIntrinsicContentSize: CGSize?
-    override open var intrinsicContentSize: CGSize {
-        defaultIntrinsicContentSize ?? super.intrinsicContentSize
```

Comment:
Don’t know how I feel about the intrinsicContentSize, I think it is adding more options to change the layout, and we should try to follow the same strategy always. Also, the customer can override the intrinsic content size if they want. WDYT ?

Reply:
You can always override `setupLayout`, not call super and not set any constraints if you wanna use just `intrinsicContentSize` :)